### PR TITLE
Fix #805: Add timestamps to proximity anti-fraud check

### LINKED ISSUE #805: Add timestamps to proximity anti-fraud check
`OperationApproveRequest` should contain two timestamps.

- when OTP was delivered to the application,
- when OTP was appended to the request

At the very beginning, it is only for logging purposes. Later on back end server may better estimate the correction for a possible time shift.

A follow-up to #743

### PR DESCRIPTION
**Pull request by banterCZ**
```json
{
  "requestObject": {
    "id": "string",
    "data": "string",
    "proximityCheck": {
      "otp": "string",
      "type": "QR_CODE",
      "timestampRequested": "2023-07-18T07:56:49.516Z",
      "timestampSigned": "2023-07-18T07:56:49.516Z"
    }
  }
}
```

### CODE REVIEW

**Comment by kober32** [comment]
Shouldn't it be `timestampCalculated` according to the documentation provided by the RB? Which states:

```
T_DL_ACTIVATED ~ the system time when the Deep Link message with the proximity OTP was delivered to the application,
T_REQ_SIGNED ~ the system time right before the cryptographic authentication code was appended to
the request.
```

**Review by romanstrobl — APPROVED** [review]
Looks OK.

**Review by S-code652 — APPROVED** [review]

File: mtoken-model/src/main/java/com/wultra/security/powerauth/lib/mtoken/model/request/OperationApproveRequest.java [hunk 1]
```diff
@@ -35,7 +39,40 @@ public class OperationApproveRequest {
...
-    private String proximityCheckOtp;
+    @Schema(description = "Optional proximity check data." )
+    private ProximityCheck proximityCheck;
+
+    public Optional<ProximityCheck> getProximityCheck() {
+        return Optional.ofNullable(proximityCheck);
+    }
+
+    @Data
+    public static class ProximityCheck {
```
Comment (petrdvorak) [inline mtoken-model/src/main/java/com/wultra/security/powerauth/lib/mtoken/model/request/OperationApproveRequest.java:52]: Not sure if this is not long enough to justify separate class file (not a bit issue, though).

**Review by petrdvorak — APPROVED** [review]
Looks OK, with one style question.

File: mtoken-model/src/main/java/com/wultra/security/powerauth/lib/mtoken/model/request/OperationApproveRequest.java (hunk 1 above)
Comment (banterCZ) [reply mtoken-model/src/main/java/com/wultra/security/powerauth/lib/mtoken/model/request/OperationApproveRequest.java:52]: The idea is to make a nicer API, not Java classes. When you look at the payload, it is obvious, what belongs together and what is related to the proximity check. I prefer `requestObject.proximityCheck.otp` to `requestObject.proximityCheckOtp`. 
```json
{
  "requestObject": {
    "id": "string",
    "data": "string",
    "proximityCheck": {
      "otp": "string",
      "type": "QR_CODE",
      "timestampReceived": "2023-07-18T07:56:49.516Z",
      "timestampSent": "2023-07-18T07:56:49.516Z"
    }
  }
}
```

**Pull request merged**